test/test_main.py

```diff
@@ -0,0 +1,81 @@
+import sys
+from collections import OrderedDict
+sys.path.append('..')
+sys.path.append('./translator/GCP_cloud_function/cloud_function')
+from unittest.mock import MagicMock, patch, call, Mock
+sys.modules['icone_translator'] =Mock()
+import main
+import icone_translator
+import os
+import json
+
+
+@patch('urllib.request')
+def test_get_ftp_file(request):
+    test_url='fake url'
+    test_output=main.get_ftp_file(test_url)
+    request.urlopen.assert_called_with(test_url)
+    assert True
+
+
+@patch('google.cloud.pubsub_v1.PublisherClient')
+@patch.object(main, 'get_ftp_url')
```

Comment:
Since you're setting the `main.xxx = MagicMock()` below, we don't need to `@patch.object` these in, or include them as parameters.

Reply:
This is case where it is more complicated than it seems. If we remove the patch.object line, then the main.xxx = MagicMock() line will overwrite the actual function from main, in all contexts. If we keep the patch.object line, it only mocks the function within that unit test.